### PR TITLE
Update objective user guide

### PR DESCRIPTION
**Pull request by AdrianSosic**
This PR updates the objective user guide and explains the new `SingleTargetObjective` and `DesirabilityObjective` classes.

### CODE REVIEW

**Comment by AdrianSosic** [comment]
Hi @AVHopp, I think this is one for you. I keep getting an error in the pipeline but because we cannot see the error messages, I have no clue what's going wrong. Have tried to reproduce locally, fixed some of the errors, but now the build runs on my machine. So have basically no way to investigate.

I think we really need to fix the following things in the doc script soon, because the current process just takes up waaaaaay to much time and nerves:
* The build needs to reuse files from previous builds, which will speed up generation tremendously
* Auto-trigger sphinx feature
* Pipe error message to the process output

But these are things for another PR. Right now, let's see if you can spot the problem 🙃

File: README.md
Comment (Scienfitz) [inline README.md:75]: detail unnecessary for the README and belongs in user guide

File: README.md
Comment (Scienfitz) [inline README.md:81]:
```suggestion
targets, the `DesirabilityObjective` allows to define additional settings, e.g. how these targets

```

File: docs/userguide/objectives.md
Comment (Scienfitz) [inline docs/userguide/objectives.md:97]: since when do we mention planend features as part of the suerguide? In principle possible, but I seem to remember that we dont do that in the code or user guides

One reason is that the name of this hasnt even been decidede as it depends on what. particular techology is implemented behind it

**Review by Scienfitz — CHANGES_REQUESTED** [review]

File: docs/userguide/objectives.md
Comment (AdrianSosic) [reply docs/userguide/objectives.md:97]: No worries, happy to remove it. That was just my attempt to move that original box somewhere that said `We are actively working on adding more objective modes for multiple targets.`. Do you want to keep that info somewhere in one form or the other, or drop altogether?

Comment (AdrianSosic) [reply docs/userguide/objectives.md]: Simply removed it for now so that you can simply resolve if that is your preferred choice 👍🏼

File: README.md
Comment (AdrianSosic) [reply README.md]: Move that part (and rewrote slightly to fit into the new context). Resolve if happy

File: README.md
Comment (AdrianSosic) [reply README.md]: Check if the new version is OK

File: docs/userguide/objectives.md
Comment (Scienfitz) [reply docs/userguide/objectives.md]: just keep it vague (like the original note, why did you see need to change it anyway?) and not give the impression that this concrete object is being implemented

Comment (AdrianSosic) [reply docs/userguide/objectives.md]: Ok, I just brought back the original note (now mentioning objective types instead of modes).

Comment (AdrianSosic) [reply docs/userguide/objectives.md]: ... and added one more tiny commit that improves the doctests, because that annoyed me

File: README.md
Comment (AdrianSosic) [inline README.md:75]: @AVHopp: I guess it's due to this change here, since it's currently pointing to a not-yet-existing page. We've had that problem in the past and that was one of the reasons for the force option, but I can't remember what was the final decision how to handle it and why it no longer works even though we encountered that situation already before

Comment (AVHopp) [reply README.md:75]: Guess we can resolve since we agreed how to proceed with this point?

File: tests/docs/test_docs.py
Comment (AVHopp) [inline tests/docs/test_docs.py:18]: need

File: tests/docs/test_docs.py
Comment (AVHopp) [inline tests/docs/test_docs.py:21]: Could you explain why this change was necessary?

**Review by AVHopp — COMMENTED** [review]
Just some minor first comments. Will review the actual user guide once the renaming is undone since this significantly simplifies looking at the diff.

File: README.md
Comment (AdrianSosic) [reply README.md:75]: Yes, will resolve

File: tests/docs/test_docs.py
Comment (AdrianSosic) [reply tests/docs/test_docs.py:21]: It was just really annoying to fix broken tests because the only info you get is like "test fails for example_27", because the parametrization just passes the example files coming from `rglob`. The change adds some nice IDs such that you know exactly which example failed and you can even nicely select the respective test to run in your IDE:

image

File: tests/docs/test_docs.py
Comment (AdrianSosic) [reply tests/docs/test_docs.py:18]: nope, `needs` is indeed correct. refers to `pseudocode`, not `files` (that's also how it used to be prior to the change, btw)

**Comment by AdrianSosic** [comment]
Hi @Scienfitz, @AVHopp. As discussed, I've reverted the name change and the doc building passes. For your convenience, I've published them on my fork so that you can easily have a look:
https://adriansosic.github.io/baybe-dev/index.html#

Once you accept, I can merge ... or let me know if something needs to be changed ;)

**Review by Scienfitz — APPROVED** [review]

File: docs/userguide/objective.md
Comment (AVHopp) [inline docs/userguide/objective.md:42]: I'd prefer "accept" instead of "swallow" (but can live with both).

**Review by AVHopp — APPROVED** [review]
In general gucci :) Just some minor comments, feel free to resolve as you see fit and merge.

File: docs/userguide/objective.md
Comment (AdrianSosic) [reply docs/userguide/objective.md:31]: Needed because there is python code in the admonition (i.e. nested stuff):
https://pradyunsg.me/furo/reference/admonitions/#nesting-admonitions

File: docs/userguide/objective.md
Comment (AdrianSosic) [reply docs/userguide/objective.md:9]: And in fact, we are in a sense already working on it because I am currently refactoring stuff for the scaling, which is directly connected ;) so I'd say let's keep it 🙃

File: docs/userguide/objective.md
Comment (AdrianSosic) [reply docs/userguide/objective.md:42]: changed 👍🏼

**Pull request merged**
